bbq/vm/vm.go

```diff
@@ -1281,10 +1277,6 @@ func castValueAndValueType(context *Context, targetType bbq.StaticType, value Va
 	// concrete outputs. In other places (e.g. interface conformance checks) we want to leave maps generic,
 	// so we don't substitute them.
 
-	// TODO: Substitute entitlements
-	//valueSemaType := interpreter.SubstituteMappedEntitlements(interpreter.MustSemaTypeOfValue(value))
-	//valueType = ConvertSemaToStaticType(interpreter, valueSemaType)
```

Comment:
Support for this was also removed from interpreter as well, sometime ago. This was a code that got copied over (from `Interpreter.VisitCastingExpression`) before the removal.